src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,101 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Facilitates the countdown for periodic refreshes.
+ */
+public class RefreshTimer extends TickingTimer{
```

Comment:
Ok. The refreshing of the sync refresh rate doesn't update the gui directly. However it does have a variable(refreshDurationInMinutes) that the GUI needs. I have moved everything related to the updating of refresh rate and data store to the ApiQuotaManager. Maybe it doesn't even need to go through the event bus.